File: lib/scripts/ruby/publish_gem.rb
module Ruby
  class PublishGem
    def run
      # Bundler release scripts
      # https://github.com/bundler/bundler/blob/master/lib/bundler/gem_helper.rb
      # also gemspec loading https://github.com/bundler/bundler/blob/ff4a522e8e75eb4ce5675a99698fb3df23b680be/lib/bundler.rb#L406
      out, stat = capture_command("bundle", "exec", "rake", "-T")

      if stat && out.include?("rake release[remote]")
        command("bundle", "exec", "rake", "release")
      else
        log_error("bundler/gem_tasks is required to use this script")
      end
    end
  end
end

File: internal/scripts/ruby/publish_gem.go
package CIScriptsRuby

import (
	"bytes"

	c "github.com/bcaldwell/ci-scripts/internal/CIScriptsHelpers"
)

type PublishGem struct{}

func (b *PublishGem) Run() error {
	out, err := c.CaptureCommand("bundle", "exec", "rake", "-T")
	if err != nil {
		return err
	}

	if bytes.Contains(out, []byte("rake release[remote]")) {
		c.Command("bundle", "exec", "rake", "release")
	} else {
		c.LogError("bundler/gem_tasks is required to use this script")
	}

	return nil
}
